user_tools/src/spark_rapids_pytools/resources/dev/prepackage_mgr.py

```diff
@@ -100,34 +100,38 @@ def _fetch_resources(self) -> dict:
 
         # Add RAPIDS JAR as dependency
         if self.tools_jar:
-            # copy from existing file. replace snapshot
-            jar_file_name = FSUtil.get_resource_name(self.tools_jar)
-            FSUtil.make_dirs(self.dest_dir)
-            dest_file = FSUtil.build_path(self.dest_dir, jar_file_name)
-            shutil.copy2(self.tools_jar, dest_file)
+            # copy from existing file.
+            tools_jar_cspath = CspPath(self.tools_jar)
+            tools_jar_url = str(tools_jar_cspath)
+            jar_file_name = tools_jar_cspath.base_name()
+            print(f'Using the provided tools_jar {tools_jar_url}')
         else:
-            # get the latest tools_jar from mvn
-            rapids_url = self._get_spark_rapids_jar_url()
-            rapids_name = FSUtil.get_resource_name(rapids_url)
-            resource_uris[rapids_url] = {'name': rapids_name, 'pbar_enabled': False}
+            tools_jar_url = self._get_spark_rapids_jar_url()
+            jar_file_name = FSUtil.get_resource_name(tools_jar_url)
+        resource_uris[tools_jar_url] = {
+            'depItem': RuntimeDependency(name=jar_file_name, uri=tools_jar_url),
+            'prettyName': jar_file_name
+        }
 
         for platform in self._supported_platforms:  # pylint: disable=no-member
             config_file = FSUtil.build_full_path(self.resource_dir,
                                                  f'{platform}{self._configs_suffix}')  # pylint: disable=no-member
             platform_conf = JSONPropertiesContainer(config_file)
             dependency_list = RapidsTool.get_rapids_tools_dependencies('LOCAL', platform_conf)
             for dependency in dependency_list:
-                uri = dependency.get('uri')
-                name = FSUtil.get_resource_name(uri)
-                if uri:
-                    resource_uris[uri] = {'name': name, 'pbar_enabled': False}
-                    resource_uris[uri + '.asc'] = {'name': name + '.asc', 'pbar_enabled': False}
+                if dependency.uri:
+                    uri_str = str(dependency.uri)
+                    pretty_name = FSUtil.get_resource_name(uri_str)
+                    resource_uris[uri_str] = {
+                        'depItem': dependency,
```

Comment:
What would be the purpose of this key `depItem`? We seem to use `uri_str` and `prettyName` only.

Reply:
Currently we do not.
I left it for future improvement and improvement in the script.
For example, the class can load all the configuration of the download entry from the json/yaml. Moving forward, if we have some changes in doing some verification/checks or extend the properties of the download item, then we won't have to do major changes in that script again.